app/component/IndexPage.js

```diff
@@ -128,6 +143,19 @@ class IndexPage extends React.Component {
     const { router, match, config } = this.context;
     const { location } = match;
 
+    const currentLocation =
+      config.startSearchFromUserLocation &&
+      !this.props.origin.address &&
+      this.props.locationState?.hasLocation &&
+      this.props.locationState;
+
+    if (config.startSearchFromUserLocation && currentLocation) {
+      const originPoint = [currentLocation.lon, currentLocation.lat];
```

Comment:
This code is now executed many times. I suggest checking if currentLocation.reverseGeocoding is no longer in progress and origin and currentLocation coords are different or something like that.